pkg/handlers/s3_handler.go

```diff
@@ -221,42 +247,46 @@ func (h *s3Handler) OnAdd(ns string, vc vckv1alpha1.VolumeConfig, controllerRef
 	usedNodeNames := []string{}
 	nodeLabelKey := fmt.Sprintf("%s/%s-%s-%s", vckv1alpha1.GroupName, ns, controllerRef.Name, vc.ID)
 	for _, vckName := range vckNames {
-		err := waitForPodSuccess(podClient, vckName, ns, timeout)
+		// Wait for pod success only if resync is not enabled.
```

Comment:
I think this would probably be bad if the initial sync failed, like wrong keys are provided, etc. Do you think it makes sense to still check for the error first and then ignore if it goes above the timeout?

Reply:
I've updated the PR to check if the pod is running after the timeout when resync is enabled.